utils/staking-miner/src/main.rs

```diff
@@ -294,6 +297,32 @@ enum Solvers {
 	},
 }
 
+#[derive(Debug, Copy, Clone)]
+enum SubmissionStrategy {
+	// Only submit if at the time, we are the best.
+	OnlySubmitIfLeading,
+	// Always submit.
+	AlwaysSubmit,
+	// Submit if we are leading, or if the solution that's leading is more that the given `Perbill`
+	// better than us. This helps detect obviously fake solutions and still combat them.
+	SubmitIfClaimBetterThan(Perbill),
+}
+
+impl FromStr for SubmissionStrategy {
+	type Err = std::num::ParseFloatError;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		let res = if s == "only-submit-if-leading" {
+			Self::OnlySubmitIfLeading
+		} else if s == "always-submit" {
+			Self::AlwaysSubmit
+		} else {
+			Self::SubmitIfClaimBetterThan(Perbill::from_float(s.parse()?))
+		};
```

Comment:
this should parse something like `--if-better-than 10` (10 would be passed to `Perbill::from_precent`)